llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -2566,9 +2566,9 @@ static Value *getExpandedStep(const InductionDescriptor &ID,
     return C->getValue();
   if (auto *U = dyn_cast<SCEVUnknown>(Step))
     return U->getValue();
-  auto I = ExpandedSCEVs.find(Step);
-  assert(I != ExpandedSCEVs.end() && "SCEV must be expanded at this point");
-  return I->second;
+  auto *V = ExpandedSCEVs.lookup(Step);
```

Comment:
This change looks perfectly reasonable to me, but why "prefer" `lookup` to `find`? Both `lookup` and `find` still call `doFind` so the work done is the same, so I assume the main reason is because it makes this assertion slightly easier to write? Similarly it reduces a few characters in the second instance below.

Reply:
Yeah I noticed that when looking at the code. I guess it's just the difference between comparing iterators and comparing a default-constructed pointer with 0, and the latter seems neater for this case because the object is a simple type.